addons/portal_rating/static/src/chatter/frontend/message_model_patch.js

```diff
@@ -0,0 +1,16 @@
+import { Message } from "@mail/core/common/message_model";
+import { patch } from "@web/core/utils/patch";
+
+/** @type {import("models").Message} */
+const messagePatch = {
+    async remove({ removeFromThread = false } = {}) {
+        const data = await super.remove(...arguments);
+        if (this.thread && removeFromThread) {
+            this.thread.messages.forEach((message) => {
+                message.rating_stats = this.thread.rating_stats;
+            });
+        }
+        return data;
+    },
```

Comment:
We have recently merged a PR that covers different deletion scenarios in portal. In the case of deleting a message with a rating that affects the thread's rating stats, since for now we get the stats data from different routes in mail and portal (in mail with thread data and in portal with message data), we need to keep the data synced in all cases. This change takes care of updating the rating_stats value of the messages in the current thread, so if a message is deleted, the rating cards would be updated accordingly.

P.S. 
- When replacing the portal fetch route with the mail one, this value should only be present in the thread data.
- This should be back ported to 18.0.